src/Symfony/Component/Cache/Traits/PhpFilesTrait.php

```diff
@@ -32,6 +32,35 @@ public static function isSupported()
     }
 
     /**
+     * @return bool
+     */
+    public function prune()
+    {
+        $time = time();
+        $pruned = true;
+        $allowCompile = static::opcacheAllowCompile();
+
+        set_error_handler($this->includeHandler);
+        try {
+            foreach (new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($this->directory, \FilesystemIterator::SKIP_DOTS), \RecursiveIteratorIterator::LEAVES_ONLY, \RecursiveIteratorIterator::CATCH_GET_CHILD) as $file) {
```

Comment:
Ha, yeah, I guess I did: I guess there _shouldn't_ be unreachable children in the cache root; I'll remove it. I'll push changes shortly that address all your comments.